libnavigation-util/src/test/java/com/mapbox/navigation/utils/internal/ThreadControllerTest.kt

```diff
@@ -10,16 +9,12 @@ import org.hamcrest.CoreMatchers.`is`
 import org.junit.Assert.assertFalse
 import org.junit.Assert.assertThat
 import org.junit.Assert.assertTrue
-import org.junit.Rule
 import org.junit.Test
 
 class ThreadControllerTest {
 
-    @get:Rule
-    val coroutineRule = MainCoroutineRule()
-
     @Test
-    fun jobCountValidationNonUIScope() = coroutineRule.runBlockingTest {
+    fun jobCountValidationNonUIScope() {
```

Comment:
maybe this test should actually use coroutineRule.runBlockingTest

since it delays inside

Reply:
Nah that's what I was saying here https://github.com/mapbox/mapbox-navigation-android/pull/2839#discussion_r417496474 It actually makes the test execution slower so I reverted 😉